src/components/Toaster/Toast/Toast.tsx

```diff
@@ -0,0 +1,290 @@
+import React from 'react';
+import {block} from '../../utils/cn';
+import {Icon} from '../../Icon';
+import {Button} from '../../Button';
+import {Link} from '../../Link';
+import {CrossIcon} from '../../icons/CrossIcon';
+import {AttentionToast} from '../../icons/AttentionToast';
+import {SuccessToast} from '../../icons/SuccessToast';
+
+import './Toast.scss';
+
+const b = block('toast');
+
+const FADE_IN_LAST_ANIMATION_NAME = 'move-left';
+const FADE_OUT_LAST_ANIMATION_NAME = 'remove-height';
+
+const DEFAULT_TIMEOUT = 5000;
+
+const TITLE_ICONS = {
+    error: AttentionToast,
+    success: SuccessToast,
+};
+
+export interface ToastAction {
+    label: string;
+    onClick: VoidFunction;
+    removeAfterClick?: boolean;
+}
+
+export type ToastType = 'error' | 'success';
+
+export interface ToastGeneralProps {
+    name: string;
+    title?: string;
+    className?: string;
+    timeout?: number;
+    allowAutoHiding?: boolean;
+    content?: React.ReactNode;
+    type?: ToastType;
+    isClosable?: boolean;
+    isOverride?: boolean;
+    actions?: ToastAction[];
+}
+
+interface ToastInnerProps {
+    removeCallback: VoidFunction;
+}
+
+interface ToastProps extends ToastGeneralProps, ToastInnerProps {}
+
+enum ToastStatus {
+    creating = 'creating',
+    showingIndents = 'showing-indents',
+    showingHeight = 'showing-height',
+    hiding = 'hiding',
+    shown = 'shown',
+}
+
+type ToastStyles = {
+    height?: number;
+    position?: 'relative';
+};
+
+interface UseCloseOnTimeoutProps {
+    onClose: VoidFunction;
+    timeout?: number;
+}
+
+function useCloseOnTimeout({onClose, timeout}: UseCloseOnTimeoutProps): {
+    onMouseLeave: VoidFunction;
+    onMouseOver: VoidFunction;
+} {
```

Comment:
don't have to specify return type, typescript can infers it automatically
```suggestion
function useCloseOnTimeout({onClose, timeout}: UseCloseOnTimeoutProps) {
```